### ISSUE TITLE
MapReduce reindex fails when metrics.enabled=true - Seems to be due to StoreManager logic

### ISSUE THREAD

**Issue by zblumen** [JanusGraph/janusgraph#1958]
- Version: JanusGraph 0.4.0
- Storage Backend: Hbase
- Mixed Index Backend: NA
- Mailing list Thread URL: https://groups.google.com/forum/#!topic/janusgraph-users/9vgLUms_kvk

I Noticed that MapReduce will fail if `metric.enabled=true` in the [graph configuration](https://www.google.com/url?q=https%3A%2F%2Fdocs.janusgraph.org%2Fbasics%2Fconfiguration-reference%2F%23metrics&sa=D&sntz=1&usg=AFQjCNGGjfb9LFiyEmjNxMNb-VMEQLhaDg).  Below is an example running two graphs – only difference between the two is the `metric.enabled` configuration field.

## metrics.enabled=false – Runs as expected

```
gremlin> graphMetricsDisabled.class
==>class org.janusgraph.graphdb.database.StandardJanusGraph

gremlin> graphMetricsDisabled.getBackend().getStoreManagerClass()
==>class org.janusgraph.diskstorage.hbase.HBaseStoreManager

```
Note above, the store manager class is set to hbase 

```
gremlin> mgmt = graphMetricsDisabled.openManagement()
==>org.janusgraph.graphdb.database.management.ManagementSystem@5182d7de

gremlin> mr = new org.janusgraph.hadoop.MapReduceIndexManagement(graphMetricsDisabled)
==>org.janusgraph.hadoop.MapReduceIndexManagement@7daa1f44

gremlin> mr.updateIndex(mgmt.getGraphIndex(newIndexName), SchemaAction.REINDEX).get()
==>org.janusgraph.hadoop.compat.h2.Hadoop2CountersScanMetrics@1482f718

```
Works!

## metrics.enabled=true – Fails

```
gremlin> graphMetricsEnabled.class
==>class org.janusgraph.graphdb.database.StandardJanusGraph

gremlin> graphMetricsEnabled.getBackend().getStoreManagerClass()
==>class org.janusgraph.diskstorage.util.MetricInstrumentedStoreManager
 ```
Note above, the store manager class is not set to hbase 

```
gremlin> mgmt = graphMetricsEnabled.openManagement()
==>org.janusgraph.graphdb.database.management.ManagementSystem@3774c48

gremlin> mr = new org.janusgraph.hadoop.MapReduceIndexManagement(graphMetricsEnabled)
==>org.janusgraph.hadoop.MapReduceIndexManagement@3138b98c

gremlin> mr.updateIndex(mgmt.getGraphIndex(newIndexName), SchemaAction.REINDEX).get()
Store manager class class org.janusgraph.diskstorage.util.MetricInstrumentedStoreManageris not supported
Type ':help' or ':h' for help.
Display stack trace? [yN]y
java.lang.IllegalArgumentException: Store manager class class org.janusgraph.diskstorage.util.MetricInstrumentedStoreManageris not supported
        at org.janusgraph.hadoop.MapReduceIndexManagement.updateIndex(MapReduceIndexManagement.java:164)
        at org.janusgraph.hadoop.MapReduceIndexManagement$updateIndex.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:115)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:135)
        at Script272.run(Script272.groovy:1)
        at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine.eval(GremlinGroovyScriptEngine.java:674)
        at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine.eval(GremlinGroovyScriptEngine.java:376)
        at javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:233)
        at org.apache.tinkerpop.gremlin.groovy.engine.GremlinExecutor.lambda$eval$0(GremlinExecutor.java:266)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

Digging into the source code, it appears the issue arises at https://github.com/JanusGraph/janusgraph/blob/v0.4/janusgraph-core/src/main/java/org/janusgraph/diskstorage/Backend.java#L145

**Comment by farodin91** [comment]
Thank you for the report, we wouldn't fix this before JanusGraph 0.5. It would require a lot more rewrite before 0.5. I hope it is okay for you.

**Comment by zblumen** [comment]
@farodin91 Understood. Thank you.
